openfe/protocols/openmm_md/plain_md_methods.py

```diff
@@ -0,0 +1,464 @@
+                    maxIterations=sim_settings.minimization_steps
+                )
+                min_pdb_file = "minimized.pdb"
+                positions = simulation.context.getState(
+                    getPositions=True, enforcePeriodicBox=False
+                ).getPositions()
+                with open(shared_basepath / min_pdb_file, "w") as f:
+                    openmm.app.PDBFile.writeFile(
+                        simulation.topology, positions, file=f, keepIds=True
+                    )
+                # equilibrate
+                if verbose:
+                    logger.info("equilibrating systems")
+                simulation.context.setVelocitiesToTemperature(
+                    to_openmm(thermo_settings.temperature))
+                simulation.step(equil_steps)
+
+                # production
+                if verbose:
+                    logger.info("running production phase")
+
+                # Get the sub selection of the system to print coords for
+                selection_indices = mdtraj.Topology.from_openmm(
+                    stateA_topology).select(sim_settings.output_indices)
+
+                # Setup the reporters
+                simulation.reporters.append(XTCReporter(
+                    file=str(shared_basepath / sim_settings.output_filename),
+                    reportInterval=sim_settings.checkpoint_interval.m,
+                    atomSubset=selection_indices))
+                simulation.reporters.append(openmm.app.CheckpointReporter(
+                    file=str(shared_basepath / sim_settings.checkpoint_storage),
+                    reportInterval=sim_settings.checkpoint_interval.m))
+                simulation.reporters.append(openmm.app.StateDataReporter(
+                    str(shared_basepath / sim_settings.log_output),
+                    sim_settings.checkpoint_interval.m,
+                    step=True,
+                    time=True,
+                    potentialEnergy=True,
+                    kineticEnergy=True,
+                    totalEnergy=True,
+                    temperature=True,
+                    volume=True,
+                    density=True,
+                ))
+                t0 = time.time()
+                simulation.step(prod_steps)
+                t1 = time.time()
+                logger.info(f"Completed simulation in {t1 - t0} seconds")
+
+                nc = shared_basepath / sim_settings.output_filename
+                chk = shared_basepath / sim_settings.checkpoint_storage
+
+        finally:
+
+            if not dry:
+                del integrator, simulation
+
+        if not dry:  # pragma: no-cover
+            return {
+                'nc': nc,
+                'last_checkpoint': chk,
+            }
+        else:
+            return {'debug': {'system': stateA_system}}
+
+    def _execute(
+            self, ctx: gufe.Context, **kwargs,
+    ) -> dict[str, Any]:
+        outputs = self.run(scratch_basepath=ctx.scratch,
```

Comment:
Could you add in this wrapper please? https://github.com/OpenFreeEnergy/openfe/blob/main/openfe/protocols/openmm_afe/equil_solvation_afe_method.py#L742

One of those things that happened after you wrote this (sorry for the super delayed review!), that we unfortunately need to add everywhere now :(

Reply:
```
       log_system_probe(logging.INFO, paths=[ctx.scratch])

        with without_oechem_backend():
            outputs = self.run(scratch_basepath=ctx.scratch,
                               shared_basepath=ctx.shared)
```